edx_rest_framework_extensions/permissions.py

```diff
@@ -83,21 +90,61 @@ def has_permission(self, request, view):
         of the organizations specified in the content provider filters
         in the JWT used to authenticate.
         """
-        if _should_skip_jwt_scope_enforcement(request):
-            return True
         course_key = CourseKey.from_string(view.kwargs.get('course_id'))
         jwt_filters = decode_jwt_filters(request.auth)
-        for provider_type, filter_value in jwt_filters:
-            if provider_type == 'content_org' and filter_value == course_key.org:
+        for filter_type, filter_value in jwt_filters:
+            if filter_type == 'content_org' and filter_value == course_key.org:
                 return True
+        log.warning("Permission JwtHasContentOrgFilterForRequestedCourse not satisfied.")
         return False
 
 
-def _should_skip_jwt_scope_enforcement(request):
+class JwtHasUserFilterForRequestedUser(BasePermission):
     """
-    Returns True if either the request is not JWT authenticated or if
-    JWT scopes enforcement is disabled.
+    The JWT used to authenticate contains the appropriate user filter for the
+    requested user resource.
     """
-    if not is_jwt_authenticated(request):
-        return True
-    return not waffle.switch_is_active(SWITCH_ENFORCE_JWT_SCOPES)
+    message = 'JWT missing required user filter.'
+
+    def has_permission(self, request, view):
+        """
+        If the JWT has a user filter, verify that the filtered
+        user value matches the user in the URL.
+        """
+        user_filter = self._get_user_filter(request)
+        if not user_filter:
+            return True
+
+        allowed = user_filter == _get_username_param(request)
+        if not allowed:
+            log.warning("Permission JwtHasUserFilterForRequestedUser not satisfied for user_filter %s.", user_filter)
+        return allowed
+
+    def _get_user_filter(self, request):
+        jwt_filters = decode_jwt_filters(request.auth)
+        for filter_type, filter_value in jwt_filters:
+            if filter_type == 'user':
+                if filter_value == 'me':
```

Comment:
Do we really need to support "me" in a way that could conflict with a "me" username?